llama_index/service_context.py

```diff
@@ -90,6 +90,7 @@ def from_defaults(
         prompt_helper: Optional[PromptHelper] = None,
         embed_model: Optional[EmbedType] = "default",
         node_parser: Optional[NodeParser] = None,
+        text_splitter: Optional[TextSplitter] = None,
```

Comment:
overall LGTM. what is the main diff between text_splitter and node_parser

Reply:
good question! they are actually the same (a text splitter inherits from a node parser), but it also implements a `split_text` function. They are mainly for parsing nodes based on their general text (i,e, counting tokens, splitting by sentences, etc.)

Originally I was going to call them `TextNodeParser` and have classes like `SentenceAwareNodeParser`

But, there was some concerns that no one knows what a node parser is. So now we have the current setup lol